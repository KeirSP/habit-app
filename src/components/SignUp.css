.signUp {
    display: flex;
    flex-direction: column;
    align-items: center;
}

input {
    align-items: center;
    display: flex;
    padding: 10px;
    margin: 10px; 
}

button {
    padding: 5px;
    margin: 10px;
}