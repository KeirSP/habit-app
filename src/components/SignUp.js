import React, { Component } from 'react';
import { withRouter } from 'react-router';

class SignUp extends Component {
    constructor(props) {
        super(props);
        this.state = {
            user_name: null,
            password: null,
            first_name: null,
            surname: null,
            userID: null,
            loading: false
        }
    };

    handleChange = (event) => {
        return this.setState({[event.target.name]: event.target.value})
    }

    handleSubmit = (event) => {
     
        this.setState({loading: true})
        const data = {
            user_name: this.state.user_name,
            password: this.state.password,
            first_name: this.state.first_name,
            surname: this.state.surname,
        }
        // post the data
        fetch("http://localhost:8000/users/register", {
            method: 'POST', // *GET, POST, PUT, DELETE, etc.
            mode: 'cors', // no-cors, *cors, same-origin
            headers: {
              'Content-Type': 'application/json'
              // 'Content-Type': 'application/x-www-form-urlencoded',
            },
            body: JSON.stringify(data) // body data type must match "Content-Type" header
          })
          // then sign them in
        .then(() => fetch(`http://localhost:8000/users/${this.state.user_name}/${this.state.password}`))
        .then(response => response.json())
        .then(user => {return this.setState({userID: user})})
        //this.props.location.state.userID
        .then((e) => {this.props.history.push("/habits",  { userID: this.state.userID})})
        .catch(error => {
            console.log(error);
            this.setState({loading: false});
            alert("Invalid Sign Up")});
            

        // redirect as if they had just logged in
        event.preventDefault();
    }
    
    render() {
        return (
            <div className="signUp">
                <form className="UserAuth" onSubmit={this.handleSubmit}>
                    <input type="text" placeholder="Username" value={this.state.value} name='user_name' onChange={this.handleChange} data-test="inp-username" />
                    <input type="password" placeholder="Password" value={this.state.value} name='password' onChange={this.handleChange} />
                    <input type="text" placeholder="First Name" value={this.state.value} name='first_name' onChange={this.handleChange} data-test="inp-firstName" />
                    <input type="text" placeholder="Surname" value={this.state.value} name='surname' onChange={this.handleChange} data-test="inp-surname"/>
                    <button disabled={this.state.loading} className="button" type="submit">
                    {this.state.loading && (
                        <i
                        className="fa fa-refresh fa-spin"
                        style={{ marginRight: "5px" }}
                        />
                    )}
                    {this.state.loading && <span>Creating Account</span>}
                    {!this.state.loading && <span>Create Account</span>}
                    </button>
                </form>
            </div>
        );
    }
}

export default memoryRouter(SignUp);