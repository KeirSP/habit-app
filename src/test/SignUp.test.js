import React from 'react';
import ReactDOM from 'react-dom'
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import SignUp from '../components/SignUp'
import { MemoryRouter } from 'react-router-dom';


const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-test="${val}"]`);
  }

  
describe('<SignUp />', () => {
    it('mounts to DOM', () => {
        const div = document.createElement('div');
        ReactDOM.render(
        <MemoryRouter>    
            <SignUp />
        </MemoryRouter>
        , div);
        ReactDOM.unmountComponentAtNode(div);
    });
  
    let wrapper;

    beforeEach(() => wrapper = shallow(<SignUp.WrappedComponent />));
    it('Renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
  
    it('should render a <div />', () => {
        expect(wrapper.find('div').length).toEqual(1);
    });

    it('should render a <input />', () => {
        expect(wrapper.find('input').length).toEqual(4);
    });
  
    it('should render a <buton />', () => {
        expect(wrapper.find('button').length).toEqual(1);
    });

    it('should render a <form />', () => {
        expect(wrapper.find('form').length).toEqual(1);
    });
  
  })

describe('Test case for testing signUp',() =>{
    let wrapper;

    it('password check',()=>{
        wrapper = shallow(<SignUp/>);
        wrapper.find('input[type="password"]').simulate('change', {target: {name: 'password', value: 'susan123'}});
        expect(wrapper.state('password')).toEqual('susan123');
    });

    it('username check',()=>{
        wrapper = shallow(<SignUp/>);
        wrapper.find('[data-test="inp-username"]').simulate('change', {target: {name: 'user_name', value: 'susan'}});
        expect(wrapper.state('user_name')).toEqual('susan');
    });

    it('first name check',()=>{
        wrapper = shallow(<SignUp/>);
        wrapper.find('[data-test="inp-firstName"]').simulate('change', {target: {name: 'first_name', value: 'susan'}});
        expect(wrapper.state('first_name')).toEqual('susan');
    });

    it('surname check',()=>{
        wrapper = shallow(<SignUp/>);
        wrapper.find('[data-test="inp-surname"]').simulate('change', {target: {name: 'surname', value: 'missaglia'}});
        expect(wrapper.state('surname')).toEqual('missaglia');
    });

    it('signup check with right data',()=>{
        wrapper = shallow(<SignUp/>);
        wrapper.find('[data-test="inp-username"]').simulate('change', {target: {name: 'user_name', value: 'susan'}});
        wrapper.find('input[type="password"]').simulate('change', {target: {name: 'password', value: 'susan123'}});
        wrapper.find('[data-test="inp-firstName"]').simulate('change', {target: {name: 'first_name', value: 'susan'}});
        wrapper.find('[data-test="inp-surname"]').simulate('change', {target: {name: 'surname', value: 'fochesatto'}});
        wrapper.find('button').simulate('click');
        expect(wrapper.state('loading')).toBe(false);
    })

})

