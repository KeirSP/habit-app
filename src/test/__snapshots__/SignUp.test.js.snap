// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> Renders correctly 1`] = `ShallowWrapper {}`;
